src/main/java/org/fundacionjala/coding/melvi/BankOcr.java

```diff
@@ -0,0 +1,266 @@
+ * checksum calculation:
+ * (d1+2*d2+3*d3+...+9*d9) mod 11 = 0
+ * So now you should also write some code that calculates the checksum for a given
+ * number, and identifies if it is a valid account number.
+ * <p>
+ * User Story 3
+ * Your boss is keen to see your results. He asks you to write out a file of your
+ * findings, one for each input file, in this format:
+ * <p>
+ * 457508000
+ * 664371495 ERR
+ * 86110??36 ILL
+ * ie the file has one account number per row. If some characters are illegible,
+ * they are replaced by a ?. In the case of a wrong checksum, or illegible number,
+ * this is noted in a second column indicating status.
+ */
+public class BankOcr {
+    static final String PATTERN0 = " _ | ||_|";
+    static final String PATTERN1 = "     |  |";
+    static final String PATTERN2 = " _  _||_ ";
+    static final String PATTERN3 = " _  _| _|";
+    static final String PATTERN4 = "   |_|  |";
+    static final String PATTERN5 = " _ |_  _|";
+    static final String PATTERN6 = " _ |_ |_|";
+    static final String PATTERN7 = " _   |  |";
+    static final String PATTERN8 = " _ |_||_|";
+    static final String PATTERN9 = " _ |_| _|";
+    public static final int TRHEE = 3;
+    public static final int INT = 3;
+    public static final int SIX = 6;
+    public static final int NINE = 9;
+    public static final int TWELVE = 12;
+    public static final int FIFTEEN = 15;
+    public static final int EIGHTEEN = 18;
+    public static final int TWENTY_ONE = 21;
+    public static final int TWENTY_FOUR = 24;
+    public static final int TWENTY_SEVEN = 27;
+    public static final int ONCE = 11;
+
+    private Map<String, String> charMap = new HashMap<String, String>() {
+        {
+            put(PATTERN0, "0");
+            put(PATTERN1, "1");
+            put(PATTERN2, "2");
+            put(PATTERN3, "3");
+            put(PATTERN4, "4");
+            put(PATTERN5, "5");
+            put(PATTERN6, "6");
+            put(PATTERN7, "7");
+            put(PATTERN8, "8");
+            put(PATTERN9, "9");
+        }
+    };
+
+    /**
+     *
+     */
+    public BankOcr() {
+
+    }
+
+
+    /**
+     * @param filePath url of file finePath
+     * @return arrayList<String> with the list of numbers
+     */
+    public ArrayList<String> scanOcr(final String filePath) {
+        ArrayList<String> resultado = new ArrayList<>();
+
+        StringBuilder char1 = new StringBuilder();
+        StringBuilder char2 = new StringBuilder();
+        StringBuilder char3 = new StringBuilder();
+        StringBuilder char4 = new StringBuilder();
+        StringBuilder char5 = new StringBuilder();
+        StringBuilder char6 = new StringBuilder();
+        StringBuilder char7 = new StringBuilder();
+        StringBuilder char8 = new StringBuilder();
+        StringBuilder char9 = new StringBuilder();
+        BufferedReader br;
+        try {
+            br = new BufferedReader(new InputStreamReader(new FileInputStream(filePath), "UTF-8"));
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Use try-with-resources or close this "BufferedReader" in a "finally" clause. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2095)